dominion-pax-exam-driver/src/main/java/org/codice/dominion/pax/exam/internal/AnnotationBasedPaxExamDriverInterpolator.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.dominion.pax.exam.internal;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Version of the interpolator which will support interpolating attributes of a given annotation as
+ * long as the attribute is prefixed with <code>"annotation."</code>.
+ */
+public class AnnotationBasedPaxExamDriverInterpolator extends PaxExamDriverInterpolator {
+  private final Annotation annotation;
+
+  private final PaxExamDriverInterpolator interpolator;
+
+  public AnnotationBasedPaxExamDriverInterpolator(
+      Annotation annotation, PaxExamDriverInterpolator interpolator) {
+    super(interpolator);
+    this.annotation = annotation;
+    this.interpolator = interpolator;
+  }
+
+  @Nullable
+  @Override
+  public String lookup(String key) {
+    String value = interpolator.lookup(key);
+
+    if (value == null) {
+      if (key.startsWith("annotation.")) {
+        final String attribute = key.substring(11);
+
+        try {
+          final Method method = annotation.annotationType().getMethod(attribute);
+          final Object obj = method.invoke(annotation);
+
+          value = Objects.toString(obj, null);
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
```

Comment:
As we all other interpolations, it will simply fallback to leaving the string as is and that means you would get "annotation.xxxx". I'll throw back a different exception and stop it right there.